hdk-rust/src/api/get_links.rs

```diff
@@ -39,7 +39,7 @@ pub fn get_links_with_options(
     base: &Address,
     link_type: LinkMatch<&str>,
     tag: LinkMatch<&str>,
```

Comment:
@willemolding I seem to recall you saying Rust's type inference foo isn't strong enough to make these two `LinkMatch<&S>` where `S: Into<String>`, right?

Reply:
(oh, and you said the most likely case was that people would be using it with literals anyway, right?)

Reply:
Yeah I spent ages trying to figure this out. You can do it using `LinkMatch<S>` where `S: Into<String>` and it works great unless you pass `LinkMatch::Any`. Then it can't infer `S` even though it doesn't need to and you have to write `LinkMatch<String>::Any` or something like that every time you want to use the `Any` variant.

I decided to go with this way where you only need to do a type conversion when calling it with a `String` which I figure is the least common case.